### ISSUE TITLE
PostgreSQL Test Error using ternary datamember for subclass

### ISSUE THREAD

**Issue by sasukevita** [linq2db/linq2db#2926]
**define**
```c#
        class TestParentClass
        {
            public string var1 { get; set; }
            public TestChildClass child { get; set; }
        }
        class TestChildClass
        {
            public string var1 { get; set; }
        }
```
**produce error**
```c#
var query = from x in db.Table
                        select new TestParentClass
                        {
                            var1 = "a",
                            child = x.Id == 0 ? new TestChildClass
                            {
                                var1 = "1"
                            } :
                            new TestChildClass
                            {
                                var1 = "0"
                            }
                        };
            query.Where(x => x.child != null).ToArray();//error
```
Error: System.NotSupportedException: 'The CLR type PGTest.Sample.TestBug+TestChildClass isn't natively supported by Npgsql or your PostgreSQL. To use it with a PostgreSQL composite you need to specify DataTypeName or to map it, please refer to the documentation.'

**temporary fix**

- use ternary datamember inside subclass member

```c#
var query = from x in db.Table
                        select new TestParentClass
                        {
                            var1 = "a",
                            child = new TestChildClass
                            {
                                var1 = x.Id == 0 ? "1" : "0"
                            }
                        };
            query.Where(x => x.child != null).ToArray();//works
```
- or using subquery inside subclass

```c#
var query = from x in db.Table
                        select new TestParentClass
                        {
                            var1 = "a",
                            child = x.Id == 0 ?
                            (from y in db.Table
                             select new TestChildClass
                             {
                                 var1 = "1"
                             }).FirstOrDefault() :
                            (from y in db.Table
                             select new TestChildClass
                             {
                                 var1 = "0"
                             }).FirstOrDefault()
                        };
            query.Where(x => x.child != null).ToArray();//works
```

### Environment details
linq2db version: 3.3.0
Database: PostgreSQL 12.1
.NET Framework: 4.7.2

**Comment by sdanyliv** [comment]
It is As Designed. Your ternary operator is not convertible to the SQL, so `Where` will fail.

**Comment by sasukevita** [comment]
when the case is return null
```c#
query = from x in db.Table
                        select new TestParentClass
                        {
                            var1 = "a",
                            child = x.Id == 0 ? null : new TestChildClass
                            {
                                var1 = "1"
                            }
                        };
```

so i must do fix like this even i dont need queryable for subclass?
```c#
var query = from x in db.Table
                        select new TestParentClass
                        {
                            var1 = "a",
                            child = x.Id == 0 ? null :
                            (from y in db.Table
                             select new TestChildClass
                             {
                                 var1 = "1"
                             }).FirstOrDefault()
                        };
```

**Comment by sdanyliv** [comment]
Better to post updated model. I do not see `Id` property and how entities are related.

**Comment by sasukevita** [comment]
the sample only required one table
```c#
[Table(Schema="Member", Name="Table")]
	public partial class Table
	{
		[PrimaryKey, NotNull] public int Id { get; set; } // integer
	}
```

**Comment by MaceWindu** [comment]
Fixed in 6.0 preview 1
